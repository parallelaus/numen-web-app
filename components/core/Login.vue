<template>
  <v-content>
    <v-container fluid>
      <v-layout justify-center>
        <v-flex xs12 sm8 md6 lg4 xl3>
          <v-card class="elevation-2">
            <v-toolbar dark class="elevation-2 blue darken-4">
              <v-toolbar-title class="title font-weight-thin text-capitalize">
                <img src="~/assets/images/numen-logo.svg" alt="Home" />
              </v-toolbar-title>
            </v-toolbar>
            <v-card-text>
              <v-form ref="form">
                <v-text-field
                  v-model="credentials.username"
                  :rules="usernameRequired"
                  prepend-icon="person"
                  label="Email Address"
                />
                <v-text-field
                  v-model="credentials.password"
                  :rules="passwordRequired"
                  prepend-icon="lock"
                  label="Password"
                  type="password"
                />
              </v-form>
            </v-card-text>
            <v-card-actions>
              <p v-if="loginError" class="pl-4 red--text font-weight-medium">
                Invalid Email and/or Password
              </p>
              <v-spacer />
              <v-btn color="primary" :loading="loading" @click="submit">
                Login
              </v-btn>
            </v-card-actions>
          </v-card>
        </v-flex>
      </v-layout>
    </v-container>
  </v-content>
</template>

<script>
export default {
  data: () => ({
    credentials: {
      username: '',
      password: ''
    },
    usernameRequired: [v => v.length > 0 || 'Please enter your email address'],
    passwordRequired: [v => v.length > 0 || 'Please enter your password'],
    loading: false,
    loginError: false
  }),
  methods: {
    async submit() {
      if (this.$refs.form.validate()) {
        this.loginError = false
        this.loading = true
        if (!(await this.$store.dispatch('user/login', this.credentials))) {
          this.loginError = true
        }
        this.loading = false
      }
    }
  }
}
</script>
