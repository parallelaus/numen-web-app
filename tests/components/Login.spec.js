import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuetify from 'vuetify'
import Login from '@/components/core/Login'

describe('Component: core/Login', () => {
  let wrapper
  beforeEach(() => {
    const localVue = createLocalVue()
    localVue.use(Vuetify)
    wrapper = shallowMount(Login, { localVue })
  })

  test('renders correctly', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })

  test('username required', () => {})
})
