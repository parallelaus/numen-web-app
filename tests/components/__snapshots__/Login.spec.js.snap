// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: core/Login renders correctly 1`] = `
<v-content-stub
  tag="main"
>
  <v-container-stub
    fluid=""
    tag="div"
  >
    <v-layout-stub
      justify-center=""
      tag="div"
    >
      <v-flex-stub
        lg4=""
        md6=""
        sm8=""
        tag="div"
        xl3=""
        xs12=""
      >
        <v-card-stub
          class="elevation-2"
          tag="div"
        >
          <v-toolbar-stub
            class="elevation-2 blue darken-4"
            dark="true"
            scrollthreshold="300"
          >
            <v-toolbar-title-stub
              class="title font-weight-thin text-capitalize"
            >
              <img
                alt="Home"
                src="~/assets/images/numen-logo.svg"
              />
            </v-toolbar-title-stub>
          </v-toolbar-stub>
           
          <v-card-text-stub>
            <v-form-stub>
              <v-text-field-stub
                label="Email Address"
                prepend-icon="person"
                rules="v => v.length > 0 || 'Please enter your email address'"
                value=""
              />
               
              <v-text-field-stub
                label="Password"
                prepend-icon="lock"
                rules="v => v.length > 0 || 'Please enter your password'"
                type="password"
                value=""
              />
            </v-form-stub>
          </v-card-text-stub>
           
          <v-card-actions-stub>
            <!---->
             
            <v-spacer-stub />
             
            <v-btn-stub
              activeclass="v-btn--active"
              color="primary"
              tag="button"
              type="button"
            >
              
              Login
            
            </v-btn-stub>
          </v-card-actions-stub>
        </v-card-stub>
      </v-flex-stub>
    </v-layout-stub>
  </v-container-stub>
</v-content-stub>
`;
